src/controllers/DashboardController.php

```diff
@@ -5,31 +5,33 @@
 use Cronbeat\Views\DashboardView;
 
 class DashboardController extends BaseController {
+    private $routeMap = [
+        'index' => 'showDashboard',
+        'process' => 'processDashboard'
+    ];
+    
     public function doRouting() {
-        // Parse the URL
-        $uri = $_SERVER['REQUEST_URI'];
-        $uri = trim($uri, '/');
-        $uri = explode('/', $uri);
-        
-        // Get the action
-        $action = isset($uri[1]) ? $uri[1] : 'index';
+        $path = $this->parsePath();
+        $action = !empty($path[0]) ? $path[0] : 'index';
         
-        // Handle form submissions
         if ($_SERVER['REQUEST_METHOD'] === 'POST') {
             $action = 'process';
         }
         
-        // Call the appropriate method
-        if (method_exists($this, $action)) {
-            $this->$action();
+        if (isset($this->routeMap[$action]) && method_exists($this, $this->routeMap[$action])) {
+            $method = $this->routeMap[$action];
+            $this->$method();
         } else {
-            $this->index();
+            $this->showDashboard();
         }
```

Comment:
@jetbrains-junie reduce cognitive complexite, juse use switch. it's no rocket science.